### PR TITLE
Adapt to SystemTags optimizations on server

### LINKED ISSUE #1728: /tags view performance
**Describe the bug**
When you have a lot of photos tagged, opening the /tags view will trigger lots of long running requests. 

**To Reproduce**
1. Upload lots of photos
2. Tag them all with multiple tags
3. Go to /tags view
4. See requests in network console

**Expected behavior**
The app should not request all photos per tag at once.

## Related PRs

- [x] https://github.com/nextcloud/server/pull/37961
- [x] https://github.com/nextcloud/server/pull/37969
- [ ] https://github.com/nextcloud/photos/pull/1770

### PR DESCRIPTION
**Pull request by blizzz**
Essentially, the tags overview fetches all info from systemtags-current, which contains all necessary data. Only on tags details view, files are actually being requested per REPORT.

No visual changes.

Did not ship compiled assets yet as WIP.

Requires
* [x] https://github.com/nextcloud/server/pull/37961
* [x] https://github.com/nextcloud/server/pull/37969

Fixes #1728

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,6 +77,10 @@
 		"node": "^16.0.0",
 		"npm": "^7.0.0 || ^8.0.0"
 	},
+	"jshintConfig": {
+		"asi": true,
+		"esversion": 9
+	},
 	"devDependencies": {
```
Comment (blizzz) [inline package.json:84]: btw, I needed those lines to make PhpStorm stop complaining :shrug:  Happy for other pointers :)

File: src/store/systemtags.js [hunk 2]
```diff
@@ -91,7 +90,7 @@ const getters = {
 	tags: state => state.tags,
 	tagsNames: state => state.names,
 	tag: state => id => state.tags[id],
-	tagId: state => name => state.names[name],
```
Comment (marcelklehr) [inline src/store/systemtags.js:94]: Nit: please leave the trailing comma in place :)

File: package.json (hunk 1 above)
Comment (marcelklehr) [reply package.json:84]: Mh. We're not using jshint, afaik

**Comment by marcelklehr** [comment]
You can remove this stuff as well then: https://github.com/nextcloud/photos/blob/master/lib/Controller/PageController.php#L164 and https://github.com/nextcloud/photos/blob/master/src/views/Tags.vue#L73

**Comment by blizzz** [comment]
> You can remove this stuff as well then: https://github.com/nextcloud/photos/blob/master/lib/Controller/PageController.php#L164 and https://github.com/nextcloud/photos/blob/master/src/views/Tags.vue#L73

Thanks for the pointer – done it is!

**Comment by blizzz** [comment]
The second yet open server PR is not mandatory as it is backwards compatible.

File: lib/Controller/PageController.php [hunk 3]
```diff
@@ -149,20 +149,6 @@ public function index(): TemplateResponse {
 
 		$this->initialState->provideInitialState('nomedia-paths', $paths);
 
-
-		$key = $user->getUID();
-		$tagCounts = $this->tagCountsCache->get($key);
```
Comment (marcelklehr) [inline lib/Controller/PageController.php:154]: AFAIK, this is set up in the constructor. Could you remove tagCountsCache and tagManager there as well?

**Comment by marcelklehr** [comment]
Testing this I get a 500 server error:

```
{"reqId":"VTll6Zviei2RiWJpKGWw","level":3,"time":"2023-05-16T13:02:58+00:00","remoteAddr":"192.168.21.5","user":"admin","app":"webdav","method":"PROPFIND","url":"/remote.php/dav/systemtags-assigned/image/","message":"OCA\\DAV\\SystemTag\\SystemTagNode::setNumberOfFiles(): Argument #1 ($numberOfFiles) must be of type int, string given, called in /var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php on line 102","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/113.0","version":"27.0.0.3","exception":{"Exception":"TypeError","Message":"OCA\\DAV\\SystemTag\\SystemTagNode::setNumberOfFiles(): Argument #1 ($numberOfFiles) must be of type int, string given, called in /var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php on line 102","Code":0,"Trace":[{"file":"/var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php","line":102,"function":"setNumberOfFiles","class":"OCA\\DAV\\SystemTag\\SystemTagNode","type":"->","args":["7"]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Tree.php","line":200,"function":"getChildren","class":"OCA\\DAV\\SystemTag\\SystemTagsInUseCollection","type":"->","args":[]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":900,"function":"getChildren","class":"Sabre\\DAV\\Tree","type":"->","args":["systemtags-assigned/image"]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":982,"function":"generatePathNodes","class":"Sabre\\DAV\\Server","type":"->","args":[["Sabre\\DAV\\PropFind"],[["Sabre\\DAV\\PropFind"],["OCA\\DAV\\SystemTag\\SystemTagsInUseCollection"]]]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":1662,"function":"getPropertiesIteratorForPath","class":"Sabre\\DAV\\Server","type":"->","args":["systemtags-assigned/image",["{http://owncloud.org/ns}id","{http://owncloud.org/ns}display-name","{http://owncloud.org/ns}user-visible","{http://owncloud.org/ns}user-assignable","{http://owncloud.org/ns}can-assign","And 2 more entries, set log level to debug to see all entries"],1]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":1647,"function":"writeMultiStatus","class":"Sabre\\DAV\\Server","type":"->","args":[["Sabre\\Xml\\Writer",[],"/remote.php/dav/",["d","s","oc","nc"],[]],["Generator"],false]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/CorePlugin.php","line":346,"function":"generateMultiStatus","class":"Sabre\\DAV\\Server","type":"->","args":[["Generator"],false]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpPropFind","class":"Sabre\\DAV\\CorePlugin","type":"->","args":[["Sabre\\HTTP\\Request"],["Sabre\\HTTP\\Response"]]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->","args":["method:PROPFIND",[["Sabre\\HTTP\\Request"],["Sabre\\HTTP\\Response"]]]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->","args":[["Sabre\\HTTP\\Request"],["Sabre\\HTTP\\Response"]]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/apps/dav/lib/Server.php","line":364,"function":"exec","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/remote.php","line":172,"args":["/var/www/html/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/html/apps/dav/lib/SystemTag/SystemTagNode.php","Line":190,"message":"OCA\\DAV\\SystemTag\\SystemTagNode::setNumberOfFiles(): Argument #1 ($numberOfFiles) must be of type int, string given, called in /var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php on line 102","exception":{},"CustomMessage":"OCA\\DAV\\SystemTag\\SystemTagNode::setNumberOfFiles(): Argument #1 ($numberOfFiles) must be of type int, string given, called in /var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php on line 102"}}
```

**Review by artonge — APPROVED** [review]
Nitpick, else nice :)

**Comment by juliushaertl** [comment]
Remaining server PR was merged

**Comment by blizzz** [comment]
> Remaining server PR was merged

Aye, this PR is next in row of that system tag series :)

**Comment by blizzz** [comment]
addressed both remarks, rebased and rebuild

**Comment by blizzz** [comment]
> Testing this with master I get a 500 server error:
> 
> ```
> {"reqId":"VTll6Zviei2RiWJpKGWw","level":3,"time":"2023-05-16T13:02:58+00:00","remoteAddr":"192.168.21.5","user":"admin","app":"webdav","method":"PROPFIND","url":"/remote.php/dav/systemtags-assigned/image/","message":"OCA\\DAV\\SystemTag\\SystemTagNode::setNumberOfFiles(): Argument #1 ($numberOfFiles) must be of type int, string given, called in /var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php on line 102","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/113.0","version":"27.0.0.3","exception":{"Exception":"TypeError","Message":"OCA\\DAV\\SystemTag\\SystemTagNode::setNumberOfFiles(): Argument #1 ($numberOfFiles) must be of type int, string given, called in /var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php on line 102","Code":0,"Trace":[{"file":"/var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php","line":102,"function":"setNumberOfFiles","class":"OCA\\DAV\\SystemTag\\SystemTagNode","type":"->","args":["7"]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Tree.php","line":200,"function":"getChildren","class":"OCA\\DAV\\SystemTag\\SystemTagsInUseCollection","type":"->","args":[]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":900,"function":"getChildren","class":"Sabre\\DAV\\Tree","type":"->","args":["systemtags-assigned/image"]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":982,"function":"generatePathNodes","class":"Sabre\\DAV\\Server","type":"->","args":[["Sabre\\DAV\\PropFind"],[["Sabre\\DAV\\PropFind"],["OCA\\DAV\\SystemTag\\SystemTagsInUseCollection"]]]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":1662,"function":"getPropertiesIteratorForPath","class":"Sabre\\DAV\\Server","type":"->","args":["systemtags-assigned/image",["{http://owncloud.org/ns}id","{http://owncloud.org/ns}display-name","{http://owncloud.org/ns}user-visible","{http://owncloud.org/ns}user-assignable","{http://owncloud.org/ns}can-assign","And 2 more entries, set log level to debug to see all entries"],1]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":1647,"function":"writeMultiStatus","class":"Sabre\\DAV\\Server","type":"->","args":[["Sabre\\Xml\\Writer",[],"/remote.php/dav/",["d","s","oc","nc"],[]],["Generator"],false]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/CorePlugin.php","line":346,"function":"generateMultiStatus","class":"Sabre\\DAV\\Server","type":"->","args":[["Generator"],false]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpPropFind","class":"Sabre\\DAV\\CorePlugin","type":"->","args":[["Sabre\\HTTP\\Request"],["Sabre\\HTTP\\Response"]]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->","args":["method:PROPFIND",[["Sabre\\HTTP\\Request"],["Sabre\\HTTP\\Response"]]]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->","args":[["Sabre\\HTTP\\Request"],["Sabre\\HTTP\\Response"]]},{"file":"/var/www/html/apps-extra/photos/vendor/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/apps/dav/lib/Server.php","line":364,"function":"exec","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->","args":[]},{"file":"/var/www/html/remote.php","line":172,"args":["/var/www/html/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/html/apps/dav/lib/SystemTag/SystemTagNode.php","Line":190,"message":"OCA\\DAV\\SystemTag\\SystemTagNode::setNumberOfFiles(): Argument #1 ($numberOfFiles) must be of type int, string given, called in /var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php on line 102","exception":{},"CustomMessage":"OCA\\DAV\\SystemTag\\SystemTagNode::setNumberOfFiles(): Argument #1 ($numberOfFiles) must be of type int, string given, called in /var/www/html/apps/dav/lib/SystemTag/SystemTagsInUseCollection.php on line 102"}}
> ```

Cannot reproduce this with latest master.

**Review by marcelklehr — APPROVED** [review]
:ship:

**Comment by blizzz** [comment]
/backport 49bc8faced7c5d97fd066f65524a0cf8ba257c66, a1a5cc15fb7e3aeacef68c3cfdfc1685f6cb4d1a, 0f77d3e34f2763422de081d4f38611100b41a7e5. 7a7fb0dae5935622bb9466e8fe5ccaaee7469c5a to stable27

**Comment by blizzz** [comment]
/backport 49bc8faced7c5d97fd066f65524a0cf8ba257c66, a1a5cc15fb7e3aeacef68c3cfdfc1685f6cb4d1a, 0f77d3e34f2763422de081d4f38611100b41a7e5, 7a7fb0dae5935622bb9466e8fe5ccaaee7469c5a to stable27

**Comment by marcelklehr** [comment]
/backport 49bc8faced7c5d97fd066f65524a0cf8ba257c66 a1a5cc15fb7e3aeacef68c3cfdfc1685f6cb4d1a,0f77d3e34f2763422de081d4f38611100b41a7e5,7a7fb0dae5935622bb9466e8fe5ccaaee7469c5a to stable27

**Comment by marcelklehr** [comment]
/backport 49bc8faced7c5d97fd066f65524a0cf8ba257c66,a1a5cc15fb7e3aeacef68c3cfdfc1685f6cb4d1a,0f77d3e34f2763422de081d4,38611100b41a7e5,7a7fb0dae5935622bb9466e8fe5ccaaee7469c5a to stable27

**Comment by blizzz** [comment]
In other words backport id doch yourself

**Pull request merged**